### PR TITLE
Fix for PytestAssertRewriteWarning

### PR DESCRIPTION
**Pull request by ogajduse**
## Issue statement
Every time GH Actions run  `pytest tests/robottelo/` we see the following warning message multiple times in one run. See [the following GitHub Actions run](https://github.com/SatelliteQE/robottelo/actions/runs/4892389046/jobs/8734088618#step:8:891)

```
PASSED
2023-05-05 10:30:21 - robottelo.collection - INFO - Processing test items to add testimony token markers
tests/robottelo/test_report.py::test_junit_timestamps[dummy_test-non_xdist-testcase] ..                                                                       [100%]
=============================== warnings summary ===============================
../../../../../opt/hostedtoolcache/Python/3.11.3/x64/lib/python3.11/site-packages/_pytest/config/__init__.py:747
  /opt/hostedtoolcache/Python/3.11.3/x64/lib/python3.11/site-packages/_pytest/config/__init__.py:747: PytestAssertRewriteWarning: Module already imported so cannot be rewritten: pytest_fixtures.component.satellite_auth
    self.import_plugin(import_spec)

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
- generated xml file: /home/runner/work/robottelo/robottelo/report_PMOyAWKpYi.xml - 2 passed, 1 warning in 0.02s
PASSED
tests/robottelo/test_ssh.py::TestSSH::test_command PASSED
```

This message is printed out as we import functions from the module that also contains pytest fixtures.
https://github.com/SatelliteQE/robottelo/blob/892ae8b23a574e5fd39b47de0cab126bac3d460f/pytest_fixtures/core/sat_cap_factory.py#L5-L7

There is also another warning which comes from https://github.com/reportportal/agent-python-pytest/.
```
/opt/hostedtoolcache/Python/3.11.3/x64/lib/python3.11/site-packages/pytest_reportportal/plugin.py:46: PytestDeprecationWarning: The hookimpl pytest_configure_node uses old-style configuration options (marks or attributes).
  Please use the pytest.hookimpl(optionalhook=True) decorator instead
   to configure the hooks.
   See https://docs.pytest.org/en/latest/deprecations.html#configuring-hook-specs-impls-using-markers
    @pytest.mark.optionalhook

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
```

## Solution
This PR :tada:
I have moved the functions out of the `satellite_auth.py` fixture module and put them into a newly created `IPAHost` and the good ol' `Satellite` host objects. The `IPAHost` object could be used as the `SSOHost` and could contain more functions and properties eventually.

### Additional enhancements
* Wrap the "ipa" cli commands as IPAHost methods.
* Add a new exception to represent errors in IPAHost.
* Put "kinit admin" in its own function.
* Use the IPAHost object properties instead of the data from ipa_data
  fixture.

I have also created a PR https://github.com/reportportal/agent-python-pytest/pull/337 in the plugin repo that fixes the deprecation.

## Note
I am open to moving some of the stuff to mixins to not bloat `hosts.py`.

### CODE REVIEW

**Comment by ogajduse** [comment]
trigger: test-robottelo
pytest: tests/foreman/ui/test_ldap_authentication.py

**Review by JacobCallahan — APPROVED** [review]
I like the idea! Can you look into the PRT failure and see if it's related?

File: tests/foreman/ui/test_ldap_authentication.py [hunk 1]
```diff
@@ -115,35 +114,33 @@ def rhsso_groups_teardown(default_sso_host):
...
-    )
+    result = default_ipa_host.execute(f'echo {settings.ipa.password} | kinit admin')
     assert result.status == 0
     test_user = gen_string('alpha')
     add_user_cmd = (
         f'echo {settings.ipa.password} | ipa user-add {test_user} --first'
         f'={test_user} --last={test_user} --password'
     )
-    result = ssh.command(cmd=add_user_cmd, hostname=settings.ipa.hostname)
+    result = default_ipa_host.execute(add_user_cmd)
```
Comment (omkarkhatavkar) [inline tests/foreman/ui/test_ldap_authentication.py:139]: there are many functions that add users, update, and delete users that can be added as part of the IPA host or as separate classes. this is non-blocking

File: pytest_fixtures/component/satellite_auth.py [hunk 2]
```diff
@@ -328,64 +333,26 @@ def enable_external_auth_rhsso(
...
+    ipa_host = IPAHost(target_sat)
+    ipa_host.enroll_idm_and_configure_external_auth()
     yield
-    disenroll_idm(target_sat)
+    ipa_host.disenroll_idm()
 
 
 @pytest.fixture(scope='module')
-def configure_realm(module_target_sat):
+def configure_realm(module_target_sat, default_ipa_host):
```
Comment (omkarkhatavkar) [inline pytest_fixtures/component/satellite_auth.py:355]: this is also an good candidate to move to the host class, what do you think?

**Review by omkarkhatavkar — APPROVED** [review]
ACK, non-blocking comments.

**Review by omkarkhatavkar — APPROVED** [review]
ACK, non-blocking comments.

**Comment by JacobCallahan** [comment]
I like @omkarkhatavkar's suggestion of using the ipa host to execute the commands instead of ssh.command. Technically the same in the end, but it's the better (and faster) way.

**Comment by ogajduse** [comment]
trigger: test-robottelo
pytest: tests/foreman/ui/test_ldap_authentication.py tests/foreman/cli/test_ldapauthsource.py

**Comment by ogajduse** [comment]
@omkarkhatavkar @JacobCallahan Done. :muscle:

**Comment by ogajduse** [comment]
trigger: test-robottelo
pytest: tests/foreman/ui/test_ldap_authentication.py tests/foreman/cli/test_ldapauthsource.py

**Comment by ogajduse** [comment]
trigger: test-robottelo
pytest: tests/foreman/ui/test_ldap_authentication.py tests/foreman/cli/test_ldapauthsource.py

**Comment by ogajduse** [comment]
#11468 has to be merged first

**Comment by ogajduse** [comment]
trigger: test-robottelo
pytest: tests/foreman/ui/test_ldap_authentication.py tests/foreman/cli/test_ldapauthsource.py -k 'not AD and not TestADAuthSource and not TestOpenLdapAuthSource and not OPENLDAP and not AD_2019'

**Comment by ogajduse** [comment]
trigger: test-robottelo
pytest: tests/foreman/ui/test_ldap_authentication.py tests/foreman/cli/test_ldapauthsource.py -k 'not TestOpenLdapAuthSource and not TestADAuthSource and not OPENLDAP'

**Comment by ogajduse** [comment]
Test failures are not related to the patches in this PR. PR is ready for review.

**Review by omkarkhatavkar — APPROVED** [review]
ACK, Nice Work !

**Review by lhellebr — APPROVED** [review]
LGTM, one comment needs a little discussion

File: robottelo/hosts.py [hunk 3]
```diff
@@ -1889,6 +1893,118 @@ def register_host_custom_repo(self, module_org, rhel_contenthost, repo_urls):
             # refresh repository metadata on the host
             rhel_contenthost.execute('subscription-manager repos --list')
 
+    def enroll_ad_and_configure_external_auth(self, ad_data):
```
[the remark this answers is not in the document]
Comment (omkarkhatavkar) [reply robottelo/hosts.py:1896]: I agree with @lhellebr this is very specific to the AD server and as we already have SSOHost, and IPAHost then why not an ADHost? But of course non blocking for this PR, we can achieve this in separate PR, based on author time.

**Pull request merged**